engines/sci/sci.cpp

```diff
@@ -715,6 +715,64 @@ void SciEngine::runGame() {
 	} while (true);
 }
 
+// Upon an SCI engine error state, this block will attempt to add additional VM engine context for users to provide.
+void SciEngine::errorString(const char *buf_input, char *buf_output, int buf_output_size) {
+	EngineState *s = _gamestate;
+	Script *sci = s->_segMan->getScriptIfLoaded(s->xs->addr.pc.getSegment());
+
+	// If script is actively loaded at the time of error.
+	if (sci) {
+		// Query the top-most stack frame even if it's not committed yet within the VM cycle.
+		const ExecStack *call = &(s->_executionStack.back());
+		Kernel *k = g_sci->getKernel();
+
+		// Too early within initialization and this may not be yet populated.
+		const reg_t regVersion = s->variables[VAR_GLOBAL][kGlobalVarVersionNew];
+		const Common::String version = (regVersion.getOffset() > 0) ? ", Version: " + s->_segMan->getString(regVersion) : "";
+
+		const char *objname = s->_segMan->getObjectName(call->sendp);
+		Common::String callType;
+		Common::String callingFunc;
+		Common::String pcStr;
+
+		switch (call->type) {
+		case EXEC_STACK_TYPE_CALL: // Normal function
+			callType = "Selector:";
+			callingFunc += Common::String::format("%s::%s", objname, k->getSelectorName(call->debugSelector).c_str());
+			pcStr = Common::String::format("%04x:%04x", PRINT_REG(call->addr.pc));
+			break;
+		case EXEC_STACK_TYPE_KERNEL: // Kernel function
+			if (call->debugKernelSubFunction == -1){
+				callType = "Kernel:";
+				callingFunc += Common::String::format("k%s(", k->getKernelName(call->debugKernelFunction).c_str());
+			} else {
+				callType = "SubKernel:";
+				callingFunc += Common::String::format("k%s(", k->getKernelName(call->debugKernelFunction, call->debugKernelSubFunction).c_str());
+			}
+			pcStr = "none";
+			break;
+		default:
+			break;
+		}
+
+		char fmtOutput[300];
+		Common::sprintf_s(fmtOutput, "Game: %s-%s%s\nRoom: %03d, Script: %03d, %s: %s(params if possible) @ pc=%s",
```

Comment:
You could use Common::String::format() here as well

Reply:
Good catch, I’ll adjust for consistency. Keep ‘‘em coming and thank you for the review!